src/main/java/nextstep/SearchTypeConverter.java

```diff
@@ -12,7 +12,7 @@ public class SearchTypeConverter implements Converter<String, SearchType> {
     @Override
     public SearchType convert(String source) {
         return Arrays.stream(SearchType.values())
-                .filter(s -> s.name().equalsIgnoreCase(source))
+                .filter(s -> s.getCriteria().equalsIgnoreCase(source))
```

Comment:
질문 주셨던 부분이네요.
이 로직은 SearchType의 criteria와 동일한지 비교를 하고 있는데요, getter를 이용해 꺼내면 서 비교하지 않고
SearchType에게 동일한 이름을 가졌는지 메시지를 보내면 Converter 객체가 직접 비교하지 않아도 됩니다.

비교 로직을 SearchType 내부로 옮기면 Converter에서는 아래와 같은 코드로 변경할 수 있을 것 같아요 
```suggestion
                .filter(s -> s.isSameCriteria(source))
```
피드백 드린 의도는 디미터 법칙을 지켰으면 해서 입니다.
자세한 내용은 [디미터 법칙](https://johngrib.github.io/wiki/law-of-demeter/)을 참고해 보시면 좋을 것 같아요